internal/convert/process_config.go

```diff
@@ -29,28 +29,29 @@ const (
-	LogPath             string   `yaml:"logPath" json:"logPath"`
-	LogDestination      string   `yaml:"logDestination,omitempty" json:"logDestination,omitempty"`
-	Name                string   `yaml:"name,omitempty" json:"name,omitempty"`
-	Port                int      `yaml:"port" json:"port"`
-	Priority            *float64 `yaml:"priority,omitempty" json:"priority,omitempty"`
-	ProcessType         string   `yaml:"processType" json:"processType"`
-	SlaveDelay          *float64 `yaml:"slaveDelay,omitempty" json:"slaveDelay,omitempty"`
-	Version             string   `yaml:"version,omitempty" json:"version,omitempty"`
-	Votes               *float64 `yaml:"votes,omitempty" json:"votes,omitempty"`
-	ArbiterOnly         *bool    `yaml:"arbiterOnly,omitempty" json:"arbiterOnly,omitempty"`
-	Disabled            bool     `yaml:"disabled" json:"disabled"`
-	Hidden              *bool    `yaml:"hidden,omitempty" json:"hidden,omitempty"`
+	AuditLogPath        string       `yaml:"auditLogPath,omitempty" json:"auditLogPath,omitempty"`
+	AuditLogDestination string       `yaml:"auditLogDestination,omitempty" json:"auditLogDestination,omitempty"`
+	AuditLogFormat      string       `yaml:"auditLogFormat,omitempty" json:"auditLogFormat,omitempty"`
+	BuildIndexes        *bool        `yaml:"buildIndexes,omitempty" json:"buildIndexes,omitempty"`
+	DBPath              string       `yaml:"dbPath,omitempty" json:"dbPath,omitempty"`
+	BindIP              *string      `yaml:"bindIp,omitempty" json:"bindIp,omitempty"`
+	BindIPAll           *bool        `yaml:"bindIpAll,omitempty" json:"bindIpAll,omitempty"`
+	FCVersion           string       `yaml:"featureCompatibilityVersion,omitempty" json:"featureCompatibilityVersion,omitempty"`
+	Hostname            string       `yaml:"hostname" json:"hostname"`
+	IPV6                *bool        `yaml:"ipv6,omitempty" json:"ipv6,omitempty"`
+	LogPath             string       `yaml:"logPath" json:"logPath"`
+	LogDestination      string       `yaml:"logDestination,omitempty" json:"logDestination,omitempty"`
+	Name                string       `yaml:"name,omitempty" json:"name,omitempty"`
+	Port                int          `yaml:"port" json:"port"`
+	Priority            *float64     `yaml:"priority,omitempty" json:"priority,omitempty"`
+	ProcessType         string       `yaml:"processType" json:"processType"`
+	SlaveDelay          *float64     `yaml:"slaveDelay,omitempty" json:"slaveDelay,omitempty"`
+	Version             string       `yaml:"version,omitempty" json:"version,omitempty"`
+	Votes               *float64     `yaml:"votes,omitempty" json:"votes,omitempty"`
+	ArbiterOnly         *bool        `yaml:"arbiterOnly,omitempty" json:"arbiterOnly,omitempty"`
+	Disabled            bool         `yaml:"disabled" json:"disabled"`
+	Hidden              *bool        `yaml:"hidden,omitempty" json:"hidden,omitempty"`
+	TLS                 *opsmngr.SSL `yaml:"tls,omitempty" json:"tls,omitempty"`
```

Comment:
you can't directly use `opsmngr.SSL` it doesn't have yaml tgas so a yaml configuration won't be serialised here

Reply:
this is also not as easy as just adding this field, you need to do a mapping between, opsmngr <-> convert, when reading from ops manager set the new fields in convert, when reading from convert set the right fields in ops manager

You may only need to update https://github.com/mongodb/mongocli/blob/023747d66b626d06dd0eaf72fab06622bab24d7b/internal/convert/process_config.go#L201-L207 to set ops manager TLS options since TLS is part of NET
When reading from ops manager you may need to update https://github.com/mongodb/mongocli/blob/023747d66b626d06dd0eaf72fab06622bab24d7b/internal/convert/process_config.go#L98